tests/EventEmitterTest.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Omnipay\Vindicia;
+
+use Guzzle\Http\Client;
+use Guzzle\Http\ClientInterface;
+use Omnipay\Vindicia\TestFramework\DataFaker;
+use Omnipay\Vindicia\TestFramework\SoapTestCase;
+use Omnipay\Vindicia\TestFramework\TestSubscriber;
+use PaymentGatewayLogger\Event\Constants;
+use PaymentGatewayLogger\Event\ErrorEvent;
+use PaymentGatewayLogger\Event\RequestEvent;
+use PaymentGatewayLogger\Event\ResponseEvent;
+use SoapFault;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class EventEmitterTest extends SoapTestCase
+{
+    /** @var ClientInterface */
+    protected $customHttpClient;
+
+    /** @var EventDispatcherInterface */
+    protected $eventDispatcher;
+
+    /**
+     * @return void
+     */
+    public function setUp()
+    {
+        date_default_timezone_set('Europe/London');
+
+        $this->faker = new DataFaker();
+
+        $this->currency = $this->faker->currency();
+        $this->amount = $this->faker->monetaryAmount($this->currency);
+        $this->customerId = $this->faker->customerId();
+        $this->card = $this->faker->card();
+        $this->paymentMethodId = $this->faker->paymentMethodId();
+        $this->transactionId = $this->faker->transactionId();
+        $this->transactionReference = $this->faker->transactionReference();
+        $this->riskScore = $this->faker->riskScore();
+
+        $this->customHttpClient = new Client('', array('redirect.disable' => true));
+        $this->eventDispatcher = $this->customHttpClient->getEventDispatcher();
+
+        $this->eventDispatcher->addSubscriber(new TestSubscriber());
+    }
+
+    /**
+     * Ensures that 'Request' and 'Response' events are emitted when issuing a request.
+     *
+     * @return void
+     */
+    public function testSuccessfulResponseEmitted()
+    {
+        $this->setMockSoapResponse('AuthorizeSuccess.xml', array(
+            'CURRENCY' => $this->currency,
+            'AMOUNT' => $this->amount,
+            'CUSTOMER_ID' => $this->customerId,
+            'CARD_FIRST_SIX' => substr($this->card['number'], 0, 6),
+            'CARD_LAST_FOUR' => substr($this->card['number'], -4),
+            'EXPIRY_MONTH' => $this->card['expiryMonth'],
+            'EXPIRY_YEAR' => $this->card['expiryYear'],
+            'CVV' => $this->card['cvv'],
+            'PAYMENT_METHOD_ID' => $this->paymentMethodId,
+            'TRANSACTION_ID' => $this->transactionId,
+            'TRANSACTION_REFERENCE' => $this->transactionReference,
+            'RISK_SCORE' => $this->riskScore
+        ));
+
+        $class = $this;
+        $this->eventDispatcher
+            ->addListener(
+                Constants::OMNIPAY_REQUEST_BEFORE_SEND,
+                /** @return void */
+                function (RequestEvent $event) use ($class) {
+                    $request = $event['request'];
+                    $class->assertInstanceOf('\Omnipay\Vindicia\Message\AuthorizeRequest', $request);
```

Comment:
would this assertion actually fire if no event was dispatched?

Reply:
This callback function is attached to an event listener. It will only be called when a request event is fired otherwise the assertion is never ran.

Reply:
sorry, i'm confused. how do you know an event will always be dispatched?

Reply:
Spoke offline; I was misunderstanding. Added unit tests to ensure events were being dispatched.